### PR TITLE
Data mover exposer diagnostic

### PR DESCRIPTION
**Pull request by Lyndon-Li**
Fix issue #8125, log diagnostic info for data mover exposers when expose timeout

### CODE REVIEW

**Comment by reasonerjt** [comment]
It would be better to clarify in the commit message that it only dumps the info of related objects in the log.

File: pkg/exposer/csi_snapshot.go [hunk 1]
```diff
@@ -308,6 +308,70 @@ func (e *csiSnapshotExposer) PeekExposed(ctx context.Context, ownerObject corev1
...
+	if err != nil {
+		vs = nil
+		diag += fmt.Sprintf("error getting backup vs %s, err: %v\n", backupVSName, err)
+	}
+
+	if pod != nil {
+		diag += kube.DiagnosePod(pod)
+
+		if pod.Spec.NodeName != "" {
+			if err := nodeagent.KbClientIsRunningInNode(ctx, ownerObject.Namespace, pod.Spec.NodeName, e.kubeClient); err != nil {
```
[the remark this answers is not in the document]
Comment (Lyndon-Li) [reply pkg/exposer/csi_snapshot.go:340]: The error returned covers all the cases that the function `KbClientIsRunningInNode` fails, some of the cases mean it fails to check the status of node-agent daemonset.
Looks like the error is missed in the diagnostic message here, I will add it

File: pkg/controller/data_download_controller.go [hunk 2]
```diff
@@ -677,6 +678,11 @@ func (r *DataDownloadReconciler) onPrepareTimeout(ctx context.Context, dd *veler
 		return
 	}
 
+	diags := strings.Split(r.restoreExposer.DiagnoseExpose(ctx, getDataDownloadOwnerObject(dd)), "\n")
+	for _, diag := range diags {
+		log.Warnf("[Diagnose DD expose %s/%s]%s", dd.Namespace, dd.Name, diag)
```
[the remark this answers is not in the document]
Comment (Lyndon-Li) [reply pkg/controller/data_download_controller.go]: Done, remove it

File: pkg/controller/data_upload_controller.go [hunk 3]
```diff
@@ -755,6 +756,11 @@ func (r *DataUploadReconciler) onPrepareTimeout(ctx context.Context, du *velerov
 			volumeSnapshotName = du.Spec.CSISnapshot.VolumeSnapshot
 		}
 
+		diags := strings.Split(ep.DiagnoseExpose(ctx, getOwnerObject(du)), "\n")
+		for _, diag := range diags {
+			log.Warnf("[Diagnose DU expose %s/%s]%s", du.Namespace, du.Name, diag)
```
[the remark this answers is not in the document]
Comment (Lyndon-Li) [reply pkg/controller/data_upload_controller.go]: Done, removed it

File: pkg/exposer/csi_snapshot.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lyndon-Li) [reply pkg/exposer/csi_snapshot.go:340]: Done

**Comment by Lyndon-Li** [comment]
> It would be better to clarify in the commit message that it only dumps the info of related objects in the log.

Done, changed in the PR description and chagne-log

**Review by ywk253100 — APPROVED** [review]

**Pull request merged**